pkg/cmd/grafana-cli/services/api_client.go

```diff
@@ -42,14 +44,22 @@ func (client *GrafanaComClient) GetPlugin(pluginId, repoUrl string) (models.Plug
 	return data, nil
 }
 
-func (client *GrafanaComClient) DownloadFile(pluginName, filePath, url string, checksum string) (content []byte, err error) {
+func (client *GrafanaComClient) DownloadFile(pluginName string, tmpFile *os.File, url string, checksum string) (err error) {
 	// Try handling url like local file path first
 	if _, err := os.Stat(url); err == nil {
-		bytes, err := ioutil.ReadFile(url)
+		f, err := os.Open(url)
 		if err != nil {
-			return nil, errutil.Wrap("Failed to read file", err)
+			return errutil.Wrap("Failed to read plugin archive", err)
 		}
-		return bytes, nil
+		_, err = io.Copy(tmpFile, f)
+		if err != nil {
+			return errutil.Wrap("Failed to copy plugin archive", err)
+		}
+		err = tmpFile.Close()
```

Comment:
I would think that it should be the responsibility of the function creating `tmpFile` to close it (via `defer`).

Reply:
Done.